sound/soc/sof/core.c

```diff
@@ -255,23 +255,22 @@ static int sof_machine_check(struct snd_sof_dev *sdev)
 	return 0;
 }
 
-static int sof_probe(struct platform_device *pdev)
+int snd_sof_device_probe(struct device *dev, struct snd_sof_pdata *plat_data)
 {
-	struct snd_sof_pdata *plat_data = dev_get_platdata(&pdev->dev);
 	struct snd_sof_dev *sdev;
 	const char *drv_name;
 	const void *mach;
 	int size;
 	int ret;
 
-	sdev = devm_kzalloc(&pdev->dev, sizeof(*sdev), GFP_KERNEL);
+	sdev = devm_kzalloc(dev, sizeof(*sdev), GFP_KERNEL);
 	if (!sdev)
 		return -ENOMEM;
 
-	dev_dbg(&pdev->dev, "probing SOF DSP device....\n");
+	dev_dbg(dev, "probing SOF DSP device....\n");
 
 	/* initialize sof device */
-	sdev->dev = &pdev->dev;
+	sdev->dev = dev;
 	sdev->parent = plat_data->dev;
```

Comment:
@plbossart this seems incorrect. dev and parent in sdev will point to the same device. We dont need parent anymore with sof-audio gone. We can just use dev->parent if needed